lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,188 @@
+import { useUniqueId } from '../../../util/idUtil';
+import { Props, propTypes, SelectedFileProps } from './types';
+
+export { Props };
+
+import '../NxFileUpload.scss';
+
+const formatSize = (size: number) => prettyBytes(size, { minimumFractionDigits: 1, maximumFractionDigits: 1 });
+
+function SelectedFile({ file, onDismiss: onDismissProp }: SelectedFileProps) {
+  // Testing on NVDA shows a need to set this as the aria-label in addition to the tooltip
+  const buttonLabel = 'Dismiss Upload';
+  const selectedFileRef = useRef(null);
+
+  function onDismiss() {
+    onDismissProp(file, selectedFileRef.current);
+  }
+
+  return (
+    <span className="nx-selected-file" ref={selectedFileRef}>
+      <span className="nx-selected-file__info">
+        <NxOverflowTooltip>
+          <span className="nx-selected-file__name">{file.name}</span>
+        </NxOverflowTooltip>
+        <span className="nx-selected-file__size">{formatSize(file.size)}</span>
+      </span>
+      <NxTooltip title={buttonLabel}>
+        <button type="button" aria-label={buttonLabel} className="nx-selected-file__dismiss-btn" onClick={onDismiss}>
+          <NxFontAwesomeIcon icon={faTimesCircle} />
+        </button>
+      </NxTooltip>
+    </span>
+  );
+}
+
+const NxFileUpload = forwardRef<HTMLDivElement, Props>(function NxFileUpload(props, ref) {
+  const {
+        className: classNameProp,
+        onChange: onChangeProp,
+        files,
+        isRequired,
+        isPristine,
+        id,
+        disabled,
+        ...attrs
+      } = props,
+      isFileSelected = !!files?.length,
+      { showValidationErrors: formShowValidationErrors } = useContext(FormAriaContext),
+      showValidationErrors = formShowValidationErrors || !isPristine,
+      showError = isRequired && showValidationErrors && !isFileSelected,
+      className = classnames('nx-multi-file-upload', classNameProp),
+      noFileMessageClassName = classnames('nx-multi-file-upload__no-file-message', {
+        'nx-multi-file-upload__no-file-message--invalid': showError
+      }),
+      inputRef = useRef<HTMLInputElement>(null),
+      inputId = useUniqueId('nx-multi-file-upload-input', id),
+      selectedFilesContainerRef = useRef<HTMLDivElement>(null),
+      validationErrorId = useUniqueId('nx-multi-file-upload-validation-error'),
+      [previousFiles, setPreviousFiles] = useState<FileList | null>(files);
+
+  function onChange(evt: FormEvent<HTMLInputElement>) {
+
+    const dataTransferObject = new DataTransfer();
+
+    // get the previous files uploaded
+    if (files) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(files));
+    }
+
+    const inputFiles = evt.currentTarget.files,
+        normalizedFiles = !inputFiles?.length ? null : inputFiles;
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    if (inputRef.current) {
+      inputRef.current.files = dataTransferObject.files;
```

Comment:
This line shouldn't be here.  Much like we don't change the `value` of a text input in its `onChange` handler, but rather pass it out to the callback and wait for the surrounding code to pass it back in via a prop change, we should do the same here.